cmd/api/src/database/auth.go

```diff
@@ -541,8 +541,13 @@ func (s *BloodhoundDB) DeleteSAMLProvider(ctx context.Context, provider model.SA
 		Model:  &provider, // Pointer is required to ensure success log contains updated fields after transaction
 	}
 
+	// Delete both the associated SSO Provider and the SAML Provider
 	return s.AuditableTransaction(ctx, auditEntry, func(tx *gorm.DB) error {
-		return CheckError(tx.WithContext(ctx).Delete(&provider))
+		if err := s.DeleteSSOProvider(ctx, int(provider.ID)); err != nil {
+			return err
+		} else {
+			return CheckError(tx.WithContext(ctx).Delete(&provider))
+		}
```

Comment:
Once we delete the SSOProvider id, the FK will cascade delete the saml provider record, making this always a no-op I think.

Reply:
In this case it won't no-op.
Since we are deleting them in a transaction, the cascade delete won't actually trigger until the transaction is committed.

This gives us a slight benefit with the audit log:
<img width="1828" alt="image" src="https://github.com/user-attachments/assets/39bdd062-3fba-43b2-a5f8-8a4a73d140df">

When deleting the SAMLProvider, we also audit log the SSOProvider deletion.

Reply:
I didn't realize we wanted to keep the individual child provider logs 🤔

Reply:
Maybe not, but I could see this being beneficial to see which SAML provider was deleted. Sort of relating to the point made that we lose the context of what the SSO Provider was in the delete, this way we have a record of both entries being deleted

Reply:
Resolving, talked with @mistahj67 and came up with a cleaner way to produce the audit logs